src/modules/claims/claims.service.ts

```diff
@@ -1354,22 +1407,109 @@ export class ClaimsService {
     );
   }
 
+  /**
+   * Verifies that credential was issued by authorized issuer
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return void. Returns "Proof Not Verified" error if VC not verified. Returns error if issuer not verified
+   */
+  async verifyVc(
+    vc: VerifiableCredential<RoleCredentialSubject>
+  ): Promise<void> {
+    const issuerDID = this._signerService.did;
+    if (!(await this._verifiableCredentialService.verify(vc))) {
+      throw new Error(ERROR_MESSAGES.PROOF_NOT_VERIFIED);
+    }
+    const role = vc.credentialSubject.role.namespace;
+    await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
+  }
+
+  /**
+   * Verifies that off-chain claim has valid proof and was issued by an authorized issuer
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return void. Returns "Invalid issuer for off-chain claim" error if issuer not verified. Returns 'The credential is invalid' error if invalid proof
+   */
+  async verifyOffChainClaim(subjectDid: string, roleNameSpace: string): Promise<void> {
+    const issuerDID = this._signerService.did;
+    const claimVerifier = new ClaimIssuerVerification(this._signerService.provider, this._didRegistry.registrySettings, this._credentialResolver, this._issuerResolver);
+    if (!(await claimVerifier.verifyIssuer(issuerDID, roleNameSpace))) {
+      throw new Error(ERROR_MESSAGES.INVALID_ISSUER_OFFCHAIN_CLAIM)
+    }
+    await claimVerifier.verifyIssuance(subjectDid, roleNameSpace)
+  }
+
+  /**
+   * Determines the verification method for a claim
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return void. Returns "Proof Not Verified" error if VC not verified. Returns error if issuer not verified
+   */
+  async verify(subjectDID: string, roleNamespace: string): Promise<void> {
```

Comment:
I think this name and description is a bit more accurate. 
```suggestion
  /**
   * Resolve a credential from storage and verify its proof/signature and its issuer's authority
   *
   * @param subjectDID The DID to try to resolve a credential for
   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc") 
   * @return void. Returns "Proof Not Verified" error if VC not verified. Returns error if issuer not verified
   */
  async  resolveCredentialAndVerify(subjectDID: string, roleNamespace: string): Promise<void> {
```

Reply:
@jrhender sounds great, thanks. I think it would be good to incorporate Ashish's new method for fetching credentials as it will simplify this code.

Reply:
@jrhender Sounds good to me!